pkg/k8s/watchers/node.go

```diff
@@ -15,140 +15,58 @@
 package watchers
 
 import (
-	"context"
-	"sync"
-
 	"github.com/cilium/cilium/pkg/comparator"
 	"github.com/cilium/cilium/pkg/k8s"
-	ciliumio "github.com/cilium/cilium/pkg/k8s/apis/cilium.io"
 	"github.com/cilium/cilium/pkg/k8s/informer"
-	"github.com/cilium/cilium/pkg/node"
 	nodeTypes "github.com/cilium/cilium/pkg/node/types"
-	"github.com/cilium/cilium/pkg/option"
 
 	v1 "k8s.io/api/core/v1"
-	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/fields"
-	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/cache"
 )
 
-var (
-	// onceNodeInitStart is used to guarantee that only one function call of
-	// NodesInit is executed.
-	onceNodeInitStart sync.Once
-)
-
-// NodesInit initializes a k8s watcher for node events belonging to the node
-// where Cilium is running.
-func (k *K8sWatcher) NodesInit(k8sClient *k8s.K8sClient) {
-	onceNodeInitStart.Do(func() {
-		nodeStore, nodeController := informer.NewInformer(
-			cache.NewListWatchFromClient(k8sClient.CoreV1().RESTClient(),
-				"nodes", v1.NamespaceAll, fields.ParseSelectorOrDie("metadata.name="+nodeTypes.GetName())),
-			&v1.Node{},
-			0,
-			cache.ResourceEventHandlerFuncs{
-				AddFunc: func(obj interface{}) {
-					var valid bool
-					if node := k8s.ObjToV1Node(obj); node != nil {
-						valid = true
-						if hasAgentNotReadyTaint(node) || !k8s.HasCiliumIsUpCondition(node) {
-							k8sClient.ReMarkNodeReady()
-						}
```

Comment:
These need to be reinstated. Maybe we break them off into their own subscriber and make sure that it runs first.